example/main.go

```diff
@@ -83,10 +84,12 @@ func ip(ctx context.Context) (string, error) {
 	}
 	req.Header.Set("X-Request-ID", httpx.RequestID(ctx))
 
-	resp, err := http.DefaultClient.Do(req)
+	retrier := retry.NewRetrier("ip", retry.DefaultBackOffOpts, func(err error) bool { return true })
```

Comment:
super minor, but it's kind of hard to figure out what this third argument is doing w/o knowing what args NewRetries takes. do you think it's worth extracting to either a package level constant `retry.RetryAnyError` or a local variable `retryAnyError`?